converter.go

```diff
@@ -401,6 +385,28 @@ func arrayToString(v driver.Value, tsmode snowflakeType, params map[string]*stri
 		return bindingValue{&res, "json", &schemaForBytes}, nil
 	} else if isSliceOfSlices(v) {
 		return bindingValue{}, errors.New("array of arrays is not supported")
+	} else if valuer, ok := v1.Interface().(driver.Valuer); ok { // alternate approach; check for db valuer satisfaction
+		value, err := valuer.Value()
+
+		if err != nil || value == nil {
+			return bindingValue{}, err
+		}
+
+		if v, ok := value.(string); ok {
+			return bindingValue{&v, "", nil}, nil
+		}
+
+		return bindingValue{}, nil
+	} else if hasStringMethod(v1) { // alternate approach; check for stringer method. Guarantees it's String() and returns string
+		method := v1.MethodByName("String")
+		result := method.Call(nil) // Call with no arguments
```

Comment:
Will this panic if it has the wrong signature?

Reply:
No; the hasStringMethod validates the IO signature

Reply:
https://github.com/snowflakedb/gosnowflake/pull/1211/files#diff-ecad48054a05ca5aecf42b278530331772747ed7211a193aacd94214cb53aad0R767

Reply:
Simplified it to use `fmt.Stringer` interface check. Resolving.